### PR TITLE
You shouldn't mess with the anime club..... they have grown in power..........

### PR DESCRIPTION
**Pull request by Kurfursten**
* Pizza roll shuriken (10): Throw for great force, or eat for 8 nutriment. You can eat them even if you're full. Only weebs can throw them or eat them.
* Dakimakura (3): Substitution no jutsu! If a weeb is holding it when he takes damage, he'll try to teleport away (prioritizing a dark place). You leave the daki behind, but non-weebs can't pick it up (probably a better proc to use for this than attack_hand). They can still drag it or polyacid it, though.
* Anime poster: Silently and instantly applies a poster to the wall, hiding you inside it, just like a real ninja. Move again to pop out and return the poster to your hand. Only weebs can use it.
* Nin/tendie Power Glove: Once every 10 seconds, blast an APC by touching it, draining it of all power and adding its charge to your glove. Only weebs can use it. They will know the power of your cursed right hand and the darkness it brings... Use the charge in your blade to recharge your sword teleport or make more shuriken.

🆑 
* rscadd: The anime club now deploys 4 new incredibly deadly tools in their arsenal.

### CODE REVIEW

File: code/datums/gamemode/role/weeaboo.dm [hunk 1]
```diff
@@ -30,3 +30,158 @@
...
+					var/datum/effect/effect/system/smoke_spread/smoke = new /datum/effect/effect/system/smoke_spread()
+					smoke.set_up(3, 0, get_turf(H))
+					smoke.start()
+					//recharge = 10 SECONDS
+					H.say("Substitution no jutsu!")
+					forceMove(get_turf(H))
+					H.forceMove(target_ground)
+					return TRUE
+				else
+					to_chat(H,"<span class='warning'>There wasn't full to teleport to!</span>")
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/datums/gamemode/role/weeaboo.dm]: There wasn't full

**Comment by MadmanMartian** [comment]
Nintendie power glove seems really powerful if it just fries an APC

File: code/datums/gamemode/role/weeaboo.dm [hunk 2]
```diff
@@ -30,3 +30,158 @@
...
+			to_chat(H,"<span class='info'>It will be ready to blast an APC in [max(0,cooldown-world.time)/10] seconds.")
+
+/obj/item/clothing/gloves/nentendiepower/Touch(atom/A, mob/living/user, prox)
+	if(world.time > cooldown)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.mind.GetRole(WEEABOO))
+				if(istype(A,/obj/machinery/power/apc))
+					var/obj/machinery/power/apc/APC = A
+					APC.cell.charge = 0
```
Comment (MadmanMartian) [inline code/datums/gamemode/role/weeaboo.dm]: use cell.use(cell.charge), please.

**Comment by Kurfursten** [comment]
It's not quite the same as frying it, but you need to attach new wires which is annoying.

I could do a less powerful version where it simply sets charging and chargecount to 0, which would knock power out for 10 seconds before it charges again. That's a lot less impactful though.

File: code/datums/gamemode/role/weeaboo.dm [hunk 3]
```diff
@@ -30,3 +30,158 @@
...
+			user.forceMove(P)
+	return P
+
+obj/structure/sign/poster/anime/New()
+	..()
+	name = "anime poster"
+	desc = "It's everybody's favorite anime."
+	icon_state = pick("animeposter1")
+
+obj/structure/sign/poster/anime/relaymove(mob/user as mob)
```
Comment (MadmanMartian) [inline code/datums/gamemode/role/weeaboo.dm]: firstly you're missing some slashes here and in the above New()

Secondly, this doesn't seem to destroy the poster should the weeb leave it. I fear for the hallways that are covered in anime posters.

**Comment by boy2mantwicethefam** [comment]
@Kurfursten Use prepickup for the dakimakura

**Comment by SGuivenne** [comment]
@Kurfursten does it have a unique sprite to distinguish it from the emagged and fucked APCs?  That'd be the important bit.

**Comment by Kurfursten** [comment]
It breaks the floor tile underneath in a cry obvious way.

**Comment by Dacendeth** [comment]
10 second cd on the power glove seems really generous considering the special katana attack is a 60 second cd.

**Comment by Kurfursten** [comment]
It's not an engineering power glove, it doesn't shoot lightning, it just disables APCs

**Comment by Gringonius** [comment]
I thought that the weeb was meant to be a bad joke of an example of how to add a new antag and stop at that.

**Comment by Kurfursten** [comment]
I agree that the original is not enough to stand, which is why it needs improvement.

**Comment by Gringonius** [comment]
I didn't expect it to be made permanent, ever. This sort of silliness feels more like adminbus material rather than a standard antag, but I have to admit that I'm completely opposed adding dakis to the server even if it's a one thing only the antag can pick up anyway.

**Comment by Kurfursten** [comment]
I didn't really feel strongly either way about the weebs when they were first announced, I don't think I even posted in the PR. But I've seen them in rounds and I think they have potential - not just me either. The other admins I've talked to have had good experiences with it. They fizzle easily but people seem to have a lot of fun with them, and we can really use the midround variety. In terms of midround antags we have only ragin mages (can only fire once and costs 50 threat) and syndie sleeper (which can only fire once - I have a PR up to change that though). It should be no surprise that most of the time when a round doesn't start with a roundstart ruleset it fizzles out into extended. We need some midround variety.

The aesthetic itself is less important to me than the fact that it is a midround, loud solo antagonist. Dakis can be pokedolls or naruto logs or whatever if that's what the people want. (+1 for dolls, rocket for log, -1 for dakis)

EDIT: Speaking of aesthetic, please check out my poll here: http://ss13.moe/index.php/poll/179

**Comment by kilozombie** [comment]
While I agree this is incredibly silly, a lot of homegrown SS13 features had silly, even dumb beginnings. I think this is a good example of a new possible creative direction.

Vampire was bus-only for the longest time on Goon, AFAIK-- some will say it wasn't explored in the right way, but it's still something that was legitimized by being given love and attention, like plenty of things in the game.

**Comment by Intigracy** [comment]
The daki should have pomf on it

**Comment by DrSnips** [comment]
I can add it to my list. Current ones are (mystery anime) and lewd clown M/F.

**Comment by Kurfursten** [comment]
Big update!!
* You can now throw shuriken one handed much more easily. If you try to throw the entire stack it will throw just one. I've played with these and they feel super smooth and nice
* Posters and dakis now working and tested. Posters actually let you see "through" the wall you're hiding on but you can't pass through it (you could originally due to a bug but it was nuts powerful). I think I'm gonna deem seeing through the wall as a feature, ninja scouting from your poster outpost. Besides you can see the poster from both sides of the wall too.
* Tearing down a poster forces the ninja out. Should he also get stunned briefly?

One thing I've noticed from testing the dakis is that they put you in maint a lot. Probably gonna want to find a good way for ninjas to get out of maint. One thing I talked about with jknpj was that instead of destroying terminals, the power glove could charge up for other powers, things like
* recharging your blade teleport
* perform lightning attack with sword
* others??
In exchange for these perks he would only drain and briefly (10 ticks) pause APC charging instead of destroying the terminal in a more longterm fashion.

Also todo: I'd like the katana to be able to "crowbar" unpowered airlocks to incentivize taking out power.

**Comment by MadmanMartian** [comment]
FIIIX

**Comment by Kurfursten** [comment]
**Another Big Update**
* Nin/tendie Glove no longer destroys terminals, just drains the cell and turns off charging for 10 ticks
* Glove takes all the cell power into itself. Alt click for a radial menu. You can choose to make a new shuriken or charge your blade's teleport. If you have the shuriken in your hand already it will add to the stack.
* Cost of shuriken is a flat 1000. Blade teleport costs 40*seconds recharged. For instance, instantly refreshing your blade cooldown costs 2400.
* Posters get named. 6 exciting designs.

File: code/datums/gamemode/role/weeaboo.dm [hunk 4]
```diff
@@ -31,3 +32,300 @@
...
+
+/obj/item/clothing/gloves/nentendiepower/Touch(atom/A, mob/living/user, prox)
+	if(!prox)
+		return ..()
+	if(world.time > cooldown)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.mind.GetRole(WEEABOO))
+				if(istype(A,/obj/machinery/power/apc))
+					var/obj/machinery/power/apc/APC = A
```
Comment (jknpj) [inline code/datums/gamemode/role/weeaboo.dm:168]: use get_cell() to get your potato

Comment (jknpj) [reply code/datums/gamemode/role/weeaboo.dm:168]: but it should be fine for now

**Pull request merged**